src/HDC1080.c

```diff
@@ -48,7 +50,14 @@ float GetTemperature(void)
 	uint8_t regaddr = HDC1080_TEMP_REG;
```

Comment:
This comment pertains to the initialize functions above. The easiest way to check for an error is just to write something like the following:

if (I2C_IF_Write(...) != SUCCESS)
{
      return FALSE;
}

Also, I think you should add a delay so the device can complete the power-up sequence.